src/mono/wasm/debugger/BrowserDebugProxy/MonoSDBHelper.cs

```diff
@@ -2475,22 +2510,58 @@ public async Task<bool> ApplyUpdates(int moduleId, string dmeta, string dil, str
             return true;
         }
 
-        public async Task<byte[][]> GetBytesFromAssemblyAndPdb(string assemblyName, CancellationToken token)
+        public async Task<bool> HasDebugInfoLoadedByRuntime(string assemblyName, CancellationToken token)
+        {
+            var assemblyId = await GetAssemblyId(assemblyName, token);
+            using var commandParamsWriter1 = new MonoBinaryWriter();
+            commandParamsWriter1.Write(assemblyId);
+            using var retDebuggerCmdReader1 = await SendDebuggerAgentCommand(CmdAssembly.HasDebugInfoLoaded, commandParamsWriter1, token);
+            return retDebuggerCmdReader1.ReadByte() == 1;
+        }
+
+        public async Task<AssemblyAndPdbData> GetDataFromAssemblyAndPdb(string assemblyName, bool ignoreJMC, CancellationToken token)
         {
+            if (!assemblyName.StartsWith("System.Private.CoreLib") && !ignoreJMC && proxy.JustMyCode && !(await HasDebugInfoLoadedByRuntime(assemblyName, token)))
+                return null;
             using var commandParamsWriter = new MonoBinaryWriter();
             byte[] assembly_buf = null;
             byte[] pdb_buf = null;
+            var pdbUncompressedSize = 0;
+            (int MajorVersion, int MinorVersion) = await GetVMVersion(token);
+
             commandParamsWriter.Write(assemblyName);
             var retDebuggerCmdReader = await SendDebuggerAgentCommand(CmdVM.GetAssemblyAndPdbBytes, commandParamsWriter, token);
             int assembly_size = retDebuggerCmdReader.ReadInt32();
             if (assembly_size > 0)
                 assembly_buf = retDebuggerCmdReader.ReadBytes(assembly_size);
+            if (MajorVersion == 2 && MinorVersion >= 64)
+                pdbUncompressedSize = retDebuggerCmdReader.ReadInt32();
             int pdb_size = retDebuggerCmdReader.ReadInt32();
             if (pdb_size > 0)
                 pdb_buf = retDebuggerCmdReader.ReadBytes(pdb_size);
-            byte[][] ret = new byte[2][];
-            ret[0] = assembly_buf;
-            ret[1] = pdb_buf;
+            AssemblyAndPdbData ret = new(assembly_buf, pdb_buf);
+
+            if (MajorVersion == 2 && MinorVersion >= 64)
+            {
+                ret = new();
+                ret.AsmBytes = assembly_buf;
+                ret.PdbBytes = pdb_buf;
+                ret.HasDebugInfo = retDebuggerCmdReader.ReadBoolean();
+                ret.PdbUncompressedSize = pdbUncompressedSize;
+                if (!ret.HasDebugInfo)
+                    return ret;
+                ret.PdbAge = retDebuggerCmdReader.ReadInt32();
```

Comment:
```suggestion
                    return ret;

                ret.PdbAge = retDebuggerCmdReader.ReadInt32();
```

Just to make it easy to see that we return early here.